src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IForEachLoopStatement.cs

```diff
@@ -1219,12 +1220,13 @@ public static void M(int[] x)
 IBlockOperation (1 statements) (OperationKind.Block, Type: null, IsInvalid) (Syntax: '{ ... }')
   IForEachLoopOperation (LoopKind.ForEach) (OperationKind.Loop, Type: null, IsInvalid) (Syntax: 'foreach (x[ ... }')
     LoopControlVariable: 
-      null
-    Collection: 
-      IConversionOperation (TryCast: False, Unchecked) (OperationKind.Conversion, Type: System.Collections.IEnumerable, IsImplicit) (Syntax: 'x')
```

Comment:
>IConversionOperation (TryCast: False, Unchecked) (OperationKind.Conversion, Type: System.Collections.IEnumerable, IsImplicit) (Syntax: 'x') [](start = 6, length = 139)

Do we understand what happened with this conversion?

Reply:
Because we set the HasErrors flag, we bail out earlier in ForEachLoopBinder.BindForEachPartsWorker, before this conversion is created.

---
In reply to: [178681842](https://github.com/dotnet/roslyn/pull/25848#discussion_r178681842) [](ancestors = 178681842)